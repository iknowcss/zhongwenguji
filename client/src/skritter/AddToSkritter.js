import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import getConfig from '../getConfig';
import mapSelectors from '../util/mapSelectors';
import { missedCards } from '../characterTest/characterTestReducer';
import { userName, auth, isLoginPending } from './skritterReducer';
import Button from '../component/Button';
import style from './AddToSkritter.module.scss';

class AddToSkritter extends Component {
  static propTypes = {
    missedCards: PropTypes.array.isRequired,
    userName: PropTypes.string,
    auth: PropTypes.string,
    isLoginPending: PropTypes.bool,
  };

  static defaultProps = {
    userName: '',
    auth: '',
    isLoginPending: false
  };

  handleClick = (e) => {
    e.preventDefault();

    const { /*missedCards,*/ auth } = this.props;
    const { skritterCharactersUrl } = getConfig();

    fetch(skritterCharactersUrl, {
      method: 'POST',
      headers: { 'X-Session': auth }
    })
      .then((result) => {
        console.log(result.status);
      })

    // console.log({
    //   skritterCharactersUrl,
    //   characters: missedCards.map(({ character }) => character),
    //   auth
    // });
  };

  render() {
    const { isLoginPending } = this.props;
    return (
      <div className={style.container}>
        <div className={style.modal}>
          <div>Add all characters to Skritter</div>
          <Button
            disabled={isLoginPending}
            className={style.button}
            onClick={this.handleClick}
          >
            {isLoginPending ? 'Authorizing...' : 'Add'}
          </Button>
        </div>
      </div>
    );
  }
}

export { AddToSkritter as Pure };

export default connect(
  mapSelectors({ missedCards, userName, auth, isLoginPending })
)(AddToSkritter);
