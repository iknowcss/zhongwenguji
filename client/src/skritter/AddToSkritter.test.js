import React from 'react';
import Renderer from 'react-test-renderer';
import { Pure as AddToSkritter } from './AddToSkritter';

describe('AddToSkritter', () => {
  let component;

  function setup(props) {
    component = Renderer.create(
      <AddToSkritter
        missedCards={[]}
        userName="iknowcss"
        auth="b2hhaQ=="
        {...props}
      />
    );
  }

  it('renders with defaults', () => {
    setup();
    expect(component.toJSON()).toMatchSnapshot();
  });
});
