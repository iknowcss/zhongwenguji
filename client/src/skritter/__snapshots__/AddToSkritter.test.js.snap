// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddToSkritter renders with defaults 1`] = `
<div
  className="container"
>
  <div
    className="modal"
  >
    <div>
      Add all characters to Skritter
    </div>
    <div
      className="loggedInMessage"
    >
      Logged in as 
      <span
        className="loggedInUserName"
      >
        iknowcss
      </span>
    </div>
    <button
      className="button button"
      disabled={false}
      onClick={[Function]}
    >
      Add to Skritter
    </button>
    <button
      className="button secondary button"
    >
      Cancel
    </button>
  </div>
</div>
`;
